pkg/logs/decoder/decoder.go

```diff
@@ -119,14 +127,48 @@ func (d *Decoder) decodeIncomingData(inBuf []byte) {
 			maxj = i + d.contentLenLimit
 		} else if inBuf[j] == '\n' {
 			d.lineBuffer.Write(inBuf[i:j])
-			d.sendLine()
-			i = j + 1 // +1 as we skip the `\n`
+			if !matchDockerHeader(d.lineBuffer, d.useDockerParser) {
```

Comment:
I think we should not do anything when it matches, i.e change the condition to:
```
else if inBuf[j] == '\n' && !matchDockerHeader(d.lineBuffer, d.useDockerParser)
```
Unless I am wrong ?

Reply:
We could probably have an interface which says when the line is complete so that we limit code duplication

Reply:
Let's try to push it to v6.12.0